src/blockchains/evm/evm-provider-wrapper.ts

```diff
@@ -324,20 +329,38 @@ export class EvmProviderWrapper {
         throw e;
       }
     }
+    const chainId = await this.getChainId();
+    const networkMeta = this.networkMetadata.find((n) => n.chain_id === Number(chainId));
+
+    let maxFee, maxPrioFee;
+    if (networkMeta) {
+      const jsonRpcProvider = new EvmJsonRpcProvider(toAddress, networkMeta.rpc_urls[0]);
+      const baseFeePerGas = await jsonRpcProvider.getBaseFeePerGas();
+      const prioFeePerGas = await jsonRpcProvider.getMaxPriorityFeePerGas();
+      maxFee = baseFeePerGas * 2 + prioFeePerGas;
+      maxPrioFee = prioFeePerGas;
+    }
+
+    // Fall back to legacy fee calculation
```

Comment:
Technically this should never be needed because if a network is enabled on the backend it should provide metadata for it as well. A lot of other things also depend on it. We can leave it this way but this will probably never happen and if `networkMeta` is undefined this method could just fail.

Reply:
Good catch. My original intention was to wrap the new fee calculation into a try-catch block, and if it fails for any reason, then fall back to the old one. Will issue a fix.